src/components/Session/WithAuthentication.js

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import AuthUserContext from './context';
+import {withFirebase} from '../Firebase';
+
+const withAuthentication = Component => {
+  class WithAuthentication extends React.Component {
+    constructor(props) {
+      super(props);
+      this.state = {
+        authUser: null
+      };
+    }
+    componentDidMount() {
+      this.listener = this.props.firebase
+        ? this.props.firebase.auth.onAuthStateChanged(authUser => {
+            authUser
+              ? this.setState({authUser})
+              : this.setState({authUser: null});
```

Comment:
Siin ei ole ternary operationit vaja, piisaks `this.setState({authUser})` või kui falsy väärtuse puhul kindlasti null'iks tahta määrata, siis `this.setState({authUser: authUser || null})`.